hmpps-kotlin-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/kotlin/clienttracking/HmppsClientTrackingConfiguration.kt

```diff
@@ -0,0 +1,75 @@
+package uk.gov.justice.hmpps.kotlin.clienttracking
+
+import com.nimbusds.jwt.SignedJWT
+import io.opentelemetry.api.trace.Span
+import org.slf4j.LoggerFactory
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication.Type.REACTIVE
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication.Type.SERVLET
+import org.springframework.boot.autoconfigure.web.servlet.ConditionalOnMissingFilterBean
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Import
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer
+import java.text.ParseException
+
+internal typealias SetTrackingDetails = Any.(String) -> Unit
+
+@Configuration
+@ConditionalOnWebApplication(type = SERVLET)
+@ConditionalOnMissingBean(name = ["clientTrackingConfiguration"])
+@Import(HmppsClientTrackingInterceptorConfiguration::class)
+class HmppsClientTrackingConfiguration(private val clientTrackingInterceptor: HmppsClientTrackingInterceptor) : WebMvcConfigurer {
+  override fun addInterceptors(registry: InterceptorRegistry) {
+    log.info("Adding application insights client tracking interceptor from hmpps-spring-boot-kotlin")
+    registry.addInterceptor(clientTrackingInterceptor)
+      .addPathPatterns(clientTrackingInterceptor.includePaths)
+      .excludePathPatterns(clientTrackingInterceptor.excludePaths)
+  }
+
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+}
+
+@Configuration
+@ConditionalOnWebApplication(type = SERVLET)
+@ConditionalOnMissingBean(name = ["clientTrackingInterceptor"])
+class HmppsClientTrackingInterceptorConfiguration {
+  @Bean
+  @ConditionalOnMissingBean
+  fun hmppsClientTrackingInterceptor() = HmppsClientTrackingInterceptor()
+}
```

Comment:
For servlet based Spring Boot apps you can define your own `HmppsClientTrackingInterceptor` bean to customise the details that are tracked.